p2panda-sync/src/cbor.rs

```diff
@@ -56,20 +54,23 @@ where
     type Item = T;
     type Error = SyncError;
 
-    fn decode(
-        &mut self,
-        src: &mut tokio_util::bytes::BytesMut,
-    ) -> Result<Option<Self::Item>, Self::Error> {
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         let reader = src.reader();
-        let result: Result<Self::Item, _> = ciborium::from_reader(reader);
+        let result: Result<Self::Item, _> = decode_cbor(reader);
         match result {
             // If we read the item, we also need to advance the underlying buffer.
             Ok(item) => Ok(Some(item)),
-            Err(ref error) => match error {
-                // Sometimes the EOF is signalled as IO error
-                ciborium::de::Error::Io(_) => Ok(None),
-                e => Err(SyncError::Codec(e.to_string())),
-            },
+            Err(ref error) => {
+                match error {
+                    DecodeError::Io(err) => {
+                        // @TODO
+                        panic!("{}", err);
+                    }
```

Comment:
I'm a little bit unsure about this part. Ideally we would just convert any Io error into a critical system error (because Io will probably not be related to bad behaviour from a remote peer), but I've stumbled over this "EOF" note. Is this still the case or is this a remaining of us troubleshooting the connection bug we had in the past?

Reply:
@sandreae

Reply:
This is an artefact which came across from a cbor codec implementation I found [here](https://github.com/vorner/tokio-serde-cbor/blob/fdd3b348f9c215466ce2785e2be9d3073c3ba19a/src/lib.rs#L135). Right now I can't remember if I observed this behaviour myself, but it wasn't anything to do with the connection bug de-bugging. If all tests pass with this removed I'd say we don't need it.

Reply:
Tests are passing and it's still mysterious to me that whole part, but that's for another PR. I removed it for now and then we tackle it next.